README.md

```diff
@@ -395,19 +395,28 @@ public class JavaParserVisitor implements CommitVisitor {
 }
 ```
 
-## Dealing with Threads
+## Accelerating your analysis
 
-If your machine has multiple cores, RepoDriller can execute the commit visitor over many threads. This is just another configuration you set in _RepositoryMining_. The _withThreads()_ lets you configure the number of threads the framework will use to process everything.
+### Threads
+RepoDriller can divide the work of analyzing a repository among multiple threads. If your machine has several cores, this can significantly improve performance. However, your CommitVisitors must be thread-safe, and your analysis must tolerate visiting commits in a relatively arbitrary order.
 
-We suggest you use threads unless your workflow must _checkout_ revisions. We only keep one copy of the repository on disk, so two threads would conflict if they both ran _checkout_.
+Here are the RepositoryMining APIs you should consider:
+- _visitorsAreThreadSafe(safe)_: Are all of your CommitVisitors thread-safe?
+- _visitorsChangeRepoState(change)_: Do any of your CommitVisitors have external effects, e.g. changing the on-disk repo state with _checkout()_?
+- _withThreads(X)_: Visit the current repo from X threads
+
+### Fast storage
+RepoDriller CommitVisitors always visit a *copy* of the repository being analyzed. By default RepoDriller stores this copy in the system-wide temporary directory. If your machine has faster storage available, e.g. RAM-based (like tmpfs or ramfs on Linux) or a partition on an SSD, try specifying this path with _setRepoTmpDir(path)_. This will be particularly helpful if your CommitVisitors must _checkout()_.
 
 ```java
 @Override
 public void execute() {
 	new RepositoryMining()
 		.in(GitRepository.singleProject("/Users/mauricioaniche/workspace/repodriller"))
-		.through(Commits.all())
-		.withThreads(3)
+		.through(Commits.all()) // But not necessarily in this order with threads.
+		.visitorsAreThreadSafe(true) // So threads are possible.
+		.visitorsChangeRepoState(true) // So each thread needs its own copy of the repo.
+		.withThreads(-1) // Pick a good number of threads for my machine.
```

Comment:
As you may have figured out already, writhThreads(-1) means "choose a good value for me". The default number of threads is 1.

On reflection, I think we can set the default number of threads to -1 ("pick a good value") and this will result in >1 threads if the client calls `visitorsAreThreadSafe()`. I conservatively assume visitors are *not* thread-safe, so if the user doesn't specify the number of threads then they'll get 1 thread.

Thoughts?

Reply:
This sounds a good option, but I'd make it an explicit method in the API. So, `withThreads(n)`, n has to be >=1. If she wants us to choose the number for her, she than calls `withThreads()`. Looks more elegant. But this doesn't need to be implemented in this PR. In this PR, I'd just make sure that n >=1.